src/go/k8s/apis/cluster.redpanda.com/v1alpha1/topic_types.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2021-2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package v1alpha1
+
+import (
+	"github.com/fluxcd/pkg/apis/meta"
+	vectorizedv1alpha1 "github.com/redpanda-data/redpanda/src/go/k8s/apis/vectorized/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TopicSpec defines the desired state of Topic
+type TopicSpec struct {
+	// Partitions is the number topic shards that is distributed across the nodes in a cluster.
+	// This cannot be decreased after topic creation.
+	// It can be increased after topic creation, but it is
+	// important to understand the consequences that has, especially for
+	// topics with semantic partitioning. When absent this will default to
+	// the Redpanda cluster configuration `default_topic_partitions`.
+	// https://docs.redpanda.com/docs/reference/cluster-properties/#default_topic_partitions
+	// https://docs.redpanda.com/docs/get-started/architecture/#partitions
+	Partitions *int `json:"partitions,omitempty"`
+	// ReplicationFactor is the number of replicas the topic should have. Must be odd value.
+	// When absent this will default to the Redpanda cluster configuration `default_topic_replications`.
+	// https://docs.redpanda.com/docs/reference/cluster-properties/#default_topic_replications
+	ReplicationFactor *int `json:"replicationFactor,omitempty"`
+	// OverwriteTopicName will change the topic name from the `metadata.name` to `OverwriteTopicName`
+	OverwriteTopicName *string `json:"overwriteTopicName,omitempty"`
+	// AdditionalConfig is free form map of any configuration option that topic can have.
+	// Examples:
+	// cleanup.policy=compact
+	// redpanda.remote.write=true
+	// redpanda.remote.read=true
+	// redpanda.remote.recovery=true
+	// redpanda.remote.delete=true
+	AdditionalConfig map[string]string `json:"additionalConfig,omitempty"`
+
+	// The referenced Redpanda Cluster
+	RedpandaClusterRef vectorizedv1alpha1.NamespaceNameRef `json:"redpandaClusterRef"`
```

Comment:
This is a ClusterRef (vectorizedv1alpha1.Cluster). We also should have a RedpandaRef (redpandav1alpha1.Redpanda) to get the config from a Redpanda resource.

Reply:
I'm ok with merging this as-is and addressing this comment in a follow-up PR

Reply:
Dynamic configuration retrieval is left in the follow-up PRs.